library/src/scala/quoted/Quotes.scala

```diff
@@ -2994,6 +2995,40 @@ trait Quotes { self: runtime.QuoteUnpickler & runtime.QuoteMatching =>
       end extension
     end MatchCaseMethods
 
+    /** Type of the definition of a type lambda taking a list of type parameters. It's return type may be a TypeLambda. */
+    type ClassInfo <: TypeRepr
+
+    /** `TypeTest` that allows testing at runtime in a pattern match if a `TypeRepr` is a `TypeLambda` */
+    given ClassInfoTypeTest: TypeTest[TypeRepr, ClassInfo]
+
+    /** Module object of `type ClassInfo`  */
+    val ClassInfo: ClassInfoModule
+
+    /** Methods of the module object `val ClassInfo` */
+    trait ClassInfoModule { this: ClassInfo.type =>
+      def unapply(x: ClassInfo): (TypeRepr, Symbol, List[TypeRepr], List[Symbol], Option[TypeRepr])
+    }
+
+    /** Makes extension methods on `ClassInfo` available without any imports */
+    given ClassInfoMethods: ClassInfoMethods
+
+    /** Extension methods of `ClassInfo` */
+    trait ClassInfoMethods:
+      extension (self: ClassInfo)
+        /** The prefix on which parents, decls, and selfType need to be rebased. */
+        def prefix: TypeRepr
```

Comment:
Is `prefix` here a different thing from `qualifier` in `TypeRef`?

Reply:
@nicolasstucki Right, although I wonder if it's not too late to name it `prefix` everywhere instead of `qualifier`, for consistency with Scala 2 and previous literature